backend/inspirehep/editor/views.py

```diff
@@ -50,9 +50,12 @@ def revert_to_revision(endpoint, pid_value):
 def get_record_and_schema(endpoint, pid_value):
     pid_type = PidStoreBase.get_pid_type_from_endpoint(endpoint)
     record = InspireRecord.get_record_by_pid_value(pid_value, pid_type)
-    return jsonify(
-        {"record": {"metadata": record}, "schema": load_schema(record["$schema"])}
-    )
+    json = {"record": {"metadata": record}, "schema": load_schema(record["$schema"])}
+
+    response = make_response(json)
+    set_headers_for_record_caching_and_concurrency(response, record)
```

Comment:
this could potentially cause a state cache, when schema is changed but not the record.